src/clusterfuzz/_internal/tests/core/metrics/logs_test.py

```diff
@@ -497,3 +533,41 @@ def test_exception(self):
     logs.error('test', exception='exception', hello='1')
     self.mock.emit.assert_called_once_with(
         logging.ERROR, 'test', exc_info='err', hello='1')
+
+
+class TestLogContextSingleton(unittest.TestCase):
+  """Tests error."""
+
+  def test_is_same(self):
+    """Test the singleton is the same instance for different module loads"""
+    from clusterfuzz._internal.base.tasks.task_rate_limiting import \
+        logs as logs_from_task_rate_limiting
+    from python.bot.startup.run_bot import logs as logs_from_run_bot
+
+    self.assertIs(logs_from_run_bot, logs_from_task_rate_limiting)
+    self.assertIs(logs_from_run_bot.log_contexts,
+                  logs_from_run_bot.LogContexts())
+    logs_from_run_bot.log_contexts.add([logs_from_run_bot.LogContextType.TASK])
+
+    assert logs_from_task_rate_limiting.log_contexts == logs_from_run_bot.log_contexts
+    logs_from_run_bot.log_contexts.clear()
+
+  def test_multi_threading(self):
+    """Test multithread"""
+
+    def incrementer():
+      from python.bot.startup.run_bot import logs as run_bot_logs
+      run_bot_logs.log_contexts.add([logs.LogContextType.TASK])
+
+    import threading
```

Comment:
Why is this imported here?

Reply:
It's just to test that the singleton is thread safe. I want to propagate the context to threads that are spawned into the context

Reply:
yep, i'm using this to spawn a thread `thread = threading.Thread(target=incrementer)`